File: benchmark/perl/substring_search.pl
use Time::HiRes qw(time);
my $text = ("abcdefghij" x 1000);
my $pattern = "defg";
my $start = time();
my ($count, $tlen, $plen) = (0, length($text), length($pattern));
for my $i (0..$tlen-$plen) {
    my $match = 1;
    for my $j (0..$plen-1) { if (substr($text,$i+$j,1) ne substr($pattern,$j,1)) { $match = 0; last; } }
    $count++ if $match;
}
my $end = time();
my $ms = ($end - $start) * 1000;
print "Count: $count
";
printf "Time: %.0fms
", $ms;

File: benchmark/php/substring_search.php
<?php
$text = str_repeat("abcdefghij", 1000);
$pattern = "defg";
$start = hrtime(true);
$count = 0; $tlen = strlen($text); $plen = strlen($pattern);
for ($i = 0; $i <= $tlen - $plen; $i++) {
    $match = true;
    for ($j = 0; $j < $plen; $j++) { if ($text[$i+$j] != $pattern[$j]) { $match = false; break; } }
    if ($match) $count++;
}
$end = hrtime(true);
$ms = ($end - $start) / 1e6;
echo "Count: " . $count . "
";
echo "Time: " . round($ms) . "ms
";
